### PR TITLE
[INT-1] On-off ramp global pg

### PR DESCRIPTION
**Pull request by Jackmin801**

### CODE REVIEW

**Comment by Jackmin801** [comment]
https://github.com/PrimeIntellect-ai/ZeroBand/pull/29/commits/23ad8e5939ef2213e2237f7fd61c82b0ea8e2ad5#diff-c038af2a144a20667c1465b18904123322a1ba9ebad0d4cf0f1a21b94773839fR269-R270

took me 4 hours to find this ;-;

File: src/zeroband/comms.py [hunk 1]
```diff
@@ -10,67 +11,11 @@
 
 
 TCPSTORE_TIMEOUT = timedelta(seconds=int(os.getenv("ZERO_BAND_GLOBAL_STORE_TIMEOUT_SECONDS", "300")))
+TCPSTORE_POLLING_INTERVAL = float(os.getenv("ZERO_BAND_GLOBAL_STORE_POLLING_INTERVAL_SECONDS", "0.1"))
```
Comment (samsja) [inline src/zeroband/comms.py:14]: can you add these to to the env var summary ?

File: tests/test_dist/conftest.py [hunk 2]
```diff
@@ -37,6 +37,17 @@ def get_random_available_port():
         return s.getsockname()[1]
 
 
+@pytest.fixture()
```
Comment (samsja) [inline tests/test_dist/conftest.py]: you might have confclit here on the port

Comment (samsja) [reply tests/test_dist/conftest.py]: Please use the `get_random_available_port_list` in `test_train.py` instead

File: tests/test_dist/test_comms.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
+import time
+import torch
+import torch.distributed as dist
+import pytest
+from zeroband.comms import ElasticDeviceMesh
+import multiprocessing as mp
+
+
+@pytest.mark.parametrize("world_size", [2, 8])
```
Comment (samsja) [inline tests/test_dist/test_comms.py:9]:
```suggestion
@pytest.mark.parametrize("world_size", [2, 4])
```
lets keep the test runnable on 4 gpus

Comment (Jackmin801) [reply tests/test_dist/test_comms.py:9]: ah the tests run in cpu right now. is that ok?

File: tests/test_dist/conftest.py (hunk 2 above)
Comment (Jackmin801) [reply tests/test_dist/conftest.py]: yea I couldnt get this to work reliably so I just hardcoded some high port numbers instead. ill just drop this function and put a todo? or u think the hardcoded ports are fine and we just make sure this test is not multiprocessed?

File: src/zeroband/comms.py (hunk 1 above)
Comment (Jackmin801) [reply src/zeroband/comms.py:14]: ye sure done

File: tests/test_dist/test_comms.py (hunk 3 above)
Comment (samsja) [reply tests/test_dist/test_comms.py:9]: yeah makes sense !

File: tests/test_dist/conftest.py (hunk 2 above)
Comment (samsja) [reply tests/test_dist/conftest.py]: yeah sure

**Review by samsja — APPROVED** [review]
LFGTM :fire:

**Pull request merged**